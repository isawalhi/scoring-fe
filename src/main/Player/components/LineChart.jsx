import React from 'react';
import {
  ResponsiveContainer,
  LineChart,
  Line,
  CartesianGrid,
  XAxis,
  YAxis,
} from 'recharts';

const Chart = ({ data }) => {
  return (
    <ResponsiveContainer width="60%" height="60%" debounce={100}>
      <LineChart
        data={data}
        margin={{ top: 20, right: 20, left: 10, bottom: 5 }}
      >
        <XAxis dataKey="createdAt" axisLine tickLine />
        <YAxis
          domain={[0, 100]}
          axisLine
          tickMargin={25}
          tickLine
          height={50}
        />
        <CartesianGrid stroke="#f5f5f5" />
        <Line type="monotone" dataKey="score" stroke="#ff7300" yAxisId={0} />
      </LineChart>
    </ResponsiveContainer>
  );
};

export default Chart;
