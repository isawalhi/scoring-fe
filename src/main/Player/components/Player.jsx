import React from 'react';
import { Card } from 'antd';
import styles from './player.module.css';
import Chart from './LineChart';

const Player = ({ scoreHistory, name, lastSeen, numberOfGames, score }) => {
  return (
    <div className={styles.playerContainer}>
      <Card title={name} bordered={false}>
        <p>Number of played games: {numberOfGames}</p>
        <p>Player score: {score}</p>
        <p>Last seen: {lastSeen}</p>
      </Card>
      <Chart data={scoreHistory} score={score} />
    </div>
  );
};

export default Player;
