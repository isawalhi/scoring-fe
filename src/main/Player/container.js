import { connect } from 'react-redux';
import find from 'lodash/find';
import last from 'lodash/last';
import * as model from '../model';
import Player from './components/Player';

const mapStateToProps = (state, ownProps) => {
  const sortedPlayers = model.getSortedPlayers(state);
  // TODO dispatch an action on click and get the id from there
  const playerId = ownProps.match.params.id;
  const playerData = find(sortedPlayers, ['name', playerId]) || {};
  const { score, scoreHistory, name, numberOfGames } = playerData;
  const lastSeen = scoreHistory && last(scoreHistory).createdAt;
  return {
    name,
    score,
    lastSeen,
    scoreHistory,
    numberOfGames,
  };
};

export default connect(mapStateToProps)(Player);
