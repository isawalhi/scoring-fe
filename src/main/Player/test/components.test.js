import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Player from '../components/Player';
import LineChart from '../components/LineChart';

it('renders <Player />', () => {
  const wrapper = shallow(<Player />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders <LineChart />', () => {
  const wrapper = shallow(<LineChart />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
