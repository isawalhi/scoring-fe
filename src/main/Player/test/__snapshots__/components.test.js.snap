// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <LineChart /> 1`] = `
<ResponsiveContainer
  debounce={100}
  height="60%"
  width="60%"
>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 10,
        "right": 20,
        "top": 20,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      axisLine={true}
      dataKey="createdAt"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickLine={true}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      axisLine={true}
      domain={
        Array [
          0,
          100,
        ]
      }
      height={50}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickLine={true}
      tickMargin={25}
      type="number"
      width={60}
      yAxisId={0}
    />
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#f5f5f5"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="score"
      dot={true}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#ff7300"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </LineChart>
</ResponsiveContainer>
`;

exports[`renders <Player /> 1`] = `
<div
  className="playerContainer"
>
  <Card
    bordered={false}
  >
    <p>
      Number of played games: 
    </p>
    <p>
      Player score: 
    </p>
    <p>
      Last seen: 
    </p>
  </Card>
  <Chart />
</div>
`;
